cpp/src/parquet/CMakeLists.txt

```diff
@@ -230,6 +230,11 @@ endif()
 set(PARQUET_SHARED_LINK_LIBS)
 set(PARQUET_SHARED_PRIVATE_LINK_LIBS)
 
+if(ARROW_USE_XSIMD)
```

Comment:
should we do it here?

Reply:
I don't know. @kou What do you think?

Reply:
Yes. We need this.

Reply:
I don't know why R compile this failed...

Reply:
Oh I got to understand this, that's because `ARROW_USE_SIMD` is false, but xsimd is included...

Reply:
This is because static is not linked with xsimd